backend/hdf5/DimensionHDF5.cpp

```diff
@@ -259,13 +268,118 @@ void SetDimensionHDF5::labels(const none_t t) {
 
 SetDimensionHDF5::~SetDimensionHDF5() {}
 
+//--------------------------------------------------------------
+// Implementation of ColumnDimensionHDF5
+//--------------------------------------------------------------
+ColumnDimensionHDF5::ColumnDimensionHDF5(const H5Group &group, ndsize_t index)
+    : DimensionHDF5(group, index)
+{
+    setType();
+    // Note: maybe having this is a bad idea, because a ColumnDimension created
+    // this way will not have the block and file
+}
+
+ColumnDimensionHDF5::ColumnDimensionHDF5(const H5Group &group, ndsize_t index,  const std::shared_ptr<IFile> &file,
+                                         const std::shared_ptr<IBlock> &block)
+    : DimensionHDF5(group, index), entity_block(block), entity_file(file)
+{
+    setType();
+}
+
+ColumnDimensionHDF5::ColumnDimensionHDF5(const H5Group &group, ndsize_t index, const std::shared_ptr<IFile> &file,
+                                         const std::shared_ptr<IBlock> &block, const DataFrame &frame,
+                                         std::vector<unsigned> col_indices)
+    :ColumnDimensionHDF5(group, index, file, block)
+{
+    std::shared_ptr<IDataFrame> idf = block->getEntity<IDataFrame>(frame.id());
+    if (!idf)
+        throw std::runtime_error("ColumnDimensionHDF5 DataFrame not found in block!");
+    if (this->group.hasGroup("data_frame"))
+        this->group.removeGroup("data_frame");
+    auto target = std::dynamic_pointer_cast<DataFrameHDF5>(idf);
+
+    setType();
+    this->group.setAttr("column_indices", col_indices); //TODO
+    this->group.createLink(target->group(), "data_frame");
+}
+
+std::vector<unsigned> ColumnDimensionHDF5::columnIndices() const {
+    std::vector<unsigned> col_indices;
+    if (group.hasAttr("column_indices")) {
+        group.getAttr("column_indices", col_indices);
+        return col_indices;
+    } else {
+        throw MissingAttr("column_indices");
+    }
+}
+
+DimensionType ColumnDimensionHDF5::dimensionType() const {
+    return DimensionType::Column;
+}
+
+std::vector<std::string> ColumnDimensionHDF5::units() const {
+    std::vector<unsigned> col_indices = columnIndices();
+    nix::DataFrame df = dataFrame();
+
+    std::vector<std::string> units;
+    std::vector<Column> cols = df.columns();
+    for (unsigned index : col_indices) {
+        units.push_back(cols[index].unit);
+    }
+    return units;
+}
+
+std::vector<std::string> ColumnDimensionHDF5::labels() const {
+    std::vector<unsigned> col_indices = columnIndices();
+    nix::DataFrame df = dataFrame();
+    return df.colName(col_indices);
+}
+
+Column ColumnDimensionHDF5::column(unsigned col_index) const {
+    std::vector<nix::Column> cols = this->dataFrame()->columns();
+    return cols[col_index];
+}
```

Comment:
This returns the Column from the referenced DataFrame at a given Index.  Is it not affected by the `columnIndices` of the ColumnDimension object in any way.  Is it a convenience function for retrieving arbitrary columns from the DataFrame?

Reply:
good point, so far this function would return a column even if this is not used in the column dimension. This issue is related to my question No 3. But actually your comment indicates that the index here should be an index in the list of used indices

Reply:
Yeah I just noticed you asked similar questions in your comment and I admit I forgot you had a comment with questions in the PR.  Sorry about that.